feat(view-transitions): initial commit (#24)
thinktecture/Thinktecture.Blazor: src/Thinktecture.Blazor.ViewTransitions/README.md

# Thinktecture.Blazor.ViewTransitions

> 

Reviewer: The usual badges are missing.

## Introduction

> This package should help you to use the ViewTransition API in your Blazor application. The package contains two ways to use the ViewTransition API.

Reviewer: ```suggestion
This package should help you to use the View Transition API in your Blazor application. The package contains two ways to use the View Transition API.
```

Reviewer: Looks like this was not applied.

If you want to know how the View Transition API works look [here](https://drafts.csswg.org/css-view-transitions/#viewtransition).

> https://github.com/patrickjahr/pazor-view-transition-api/assets/16818441/424683ed-f30c-4b14-a4db-5d770c076027

Reviewer: This URL leads to a 404 page

## Getting started

### Prerequisites

You need .NET 7.0 or newer to use this library.

[Download .NET 7](https://dotnet.microsoft.com/download/dotnet/7.0)

### Platform/Browser support

> Please note that ViewTransition API is not yet supported in all major browsers. [Here](https://caniuse.com/?search=ViewTransition%20API), you can find the current support.

Reviewer: ```suggestion
Please note that View Transition API is not yet supported in all major browsers. [Here](https://caniuse.com/mdn-api_viewtransition), you can find the current support.
```

Reviewer: Looks like this was not applied.

### Installation

You can install the package via NuGet with the Package Manager in your IDE or alternatively using the command line:

```

> dotnet add package Pazor.ViewTransitionApi

Reviewer: ```suggestion
dotnet add package Thinktecture.Blazor.ViewTransitions
```

Reviewer: Looks like this change was not applied.

```

## Usage

The package can be used in Blazor WebAssembly projects.

### Add to service collection

> To make the `IViewTransitionService` available on all pages, register it at the IServiceCollection in `Program.cs` before the host is built:

Reviewer: ```suggestion
To make the IViewTransitionService available on all pages, register it at the IServiceCollection in `Program.cs` before the host is built:
```

Service names are sometimes used with backticks, sometimes without. I think like the usage _without_ backticks is the more common one in this repository.

Reviewer: Looks like this change was not applied.

> `builder.Services.AddViewTransitionServices();`

Reviewer: ```suggestion
`builder.Services.AddViewTransitionService();`
```

Reviewer: Looks this is still plural.

### Add to Imports

To use the default `RoutingViewTransition` component on the hole app razor files, register it in the `_Imports.razor` file.

```html
@using Pazor.ViewTransitionsApi
```

### Routing

For this, you must add the component `RoutingViewTransition` to App.razor.

```html
<!-- App.razor -->
<RoutingViewTransition />

<Router AppAssembly="@typeof(App).Assembly">
    <Found Context="routeData">
        <RouteView RouteData="@routeData" DefaultLayout="@typeof(MainLayout)" />
        <FocusOnNavigate RouteData="@routeData" Selector="h1" />
    </Found>
    <NotFound>
        <PageTitle>Not found</PageTitle>
        <LayoutView Layout="@typeof(MainLayout)">
            <p role="alert">Sorry, there's nothing at this address.</p>
        </LayoutView>
    </NotFound>
</Router>
```

> ### Manual ViewTransition

Reviewer: ```suggestion
### Manual view transition
```

Reviewer: Looks like this change was not applied.

> The second option is to start the `ViewTransitionAPI` using the IViewTransitionService.

Reviewer: ```suggestion
The second option is to start the View Transition API using the IViewTransitionService.
```

Reviewer: Looks like this change was not applied.

The following steps are necessary for this:
- Add the IViewTransitionService to the component or class using DependencyInjection.

`[Inject] private IViewTransitionService _viewTransitionService { get; set; } = default!;`

> - To perform a view transition, use the method `StartViewTransitionAsync`. This takes two parameters.

Reviewer: ```suggestion
- To perform a view transition, use the method `StartViewTransitionAsync()`. This takes two parameters.
```

Reviewer: Looks like this change was not applied.

>     - The first parameter is a task. This Task specifies when the transition can be performed. That means the new view is ready, and the transition can start. Please note that the ViewTransitionAPI must first take a screenshot of the current state before the DOM is changed. The following example opens a dialog. Method `StartViewTransitionAsync`, passed as a task, first waits briefly before setting the necessary parameters.

Reviewer: ```suggestion
    - The first parameter is a task. This Task specifies when the transition can be performed. That means the new view is ready, and the transition can start. Please note that the View Transition API must first take a screenshot of the current state before the DOM is changed. The following example opens a dialog. The method `StartViewTransitionAsync()`, passed as a task, first waits briefly before setting the necessary parameters.
```

Reviewer: Looks like this change was not applied.
